Feat/projects/show public

ruby10127130 opened the pull request:
### 重構 projects 頁面與元件結構
- 移除動態路由，將 page/projects/[eventId].tsx 改為 page/projects/index.tsx
- 更新 API 呼叫邏輯，將原依賴 pathname 的 useEffect 依賴項改為 []，確保僅呼叫一次 API
- 精簡頁面判斷式，將空列表顯示邏輯拆分成 EmptyList 元件，並移動到 /components/projects/ProjectList
- 拆分 ProjectList 元件，將其拆分為 ProjectCard，並進一步拆分為 ProjectCardHeader、ProjectStatus、ProjectUserInfoBar 等小元件
- 改善頁面結構，增強可讀性與維護性

ruby10127130 committed:
feat: move /projects/[eventId] to /projects

- remove no use variables

ruby10127130 committed:
chore: improve text formatting and expression clarity
- Refined `white-space-pre-wrap` implementation for better text formatting
- Improved JSX expressions to simplify logic and improve code clarity

ruby10127130 committed:
feat(shared): add component ReportMenu

ruby10127130 committed:
feat(projects): Update project card header

- change project.startDate to project.updatedDate

- add report button

ruby10127130 committed:
refactor: restructure projects page and component hierarchy

- Removed dynamic route, changed page/projects/[eventId].tsx to page/projects/index.tsx

- Updated API call logic, changed useEffect dependency to [] to ensure the API is called only once

- Simplified page conditional logic, moved empty list display to EmptyList component under /components/projects/ProjectList

- Split ProjectList component into smaller components like ProjectCard, and further split it into ProjectCardHeader, ProjectStatus, ProjectUserInfoBar

- Improved page structure for better readability and maintainability

ruby10127130 committed:
feat(ReportMenu): add noopener in window.open

JohnsonMao commented on components/Projects/ProjectList/ProjectCard/ProjectHeader.tsx:
```diff
@@ -0,0 +1,33 @@
+          <ViewIcon />
+          9999
+        </span>
+        <ReportMenu />
```
可以考慮把 ReportMenu 的元件用 `Dropdown` 來實作，就不需要另外再實作一次元件，
`Dropdown` 元件內就實作了點擊收合相關的邏輯

例如：

```tsx
      <Dropdown>
        <Dropdown.Toggle variant="solid" className="mb-1">
          <MdMoreVert />
        </Dropdown.Toggle>
        <Dropdown.List className="top-full left-0 -mt-1 z-20">
          <Dropdown.Item className="rounded-lg text-nowrap">
            <Button
              onClick={() => window.open('https://forms.gle/NkVbDWC3eXk4P4gv7', '_blank', 'noopener')}
              className="w-full text-left px-4 text-basic-500 hover:bg-basic-100 transition"
            >
              檢舉
            </Button>
          </Dropdown.Item>
        </Dropdown.List>
      </Dropdown>
```

JohnsonMao commented on pages/projects/index.tsx:
```diff
@@ -0,0 +1,113 @@
+  }, []);
+
+  return (
+    <div className="bg-[#EEF9F9]">
+      <div className="mx-auto w-[670px] max-w-full flex flex-col gap-6 px-4 py-8 md:py-28 bg-[#EEF9F9]">
```
我看外層已經設定 `bg-[#EEF9F9]`，內層應該不用再設定一次 `bg-[#EEF9F9]`

JohnsonMao commented on pages/projects/index.tsx:
```diff
@@ -0,0 +1,113 @@
+              <Skeleton animation="wave" width="95%" height="200px" className="mx-auto" />
+            )
+          }
+          {projects.length > 0
```
如果 projects 是 null 的話，這頁會 Crash
建議多判斷 projects 是否為陣列，
例如： `Array.isArray(projects)`

ruby10127130 committed:
feat(projects): adjust page/projects structure

- hide project status before api data ready

- hide project view counts before api ready

- replace ReportMenu with shared/DropDown

- remove no use background-color class

ruby10127130 commented:
### 新增
- 修正了上述 review 提到的問題
- 在 view 上隱藏了 project view, shells, comments, favorites

JohnsonMao commented:
LGTM
